front_end/SporTogetherApp/App.tsx

```diff
@@ -1,12 +1,66 @@
-import { StatusBar } from 'expo-status-bar';
-import { StyleSheet, Text, View} from 'react-native';
-import { PaperProvider, TextInput, Button } from 'react-native-paper';
-import Login from './components/Login';
-
+import { PaperProvider } from 'react-native-paper';
+import { useEffect, useState } from 'react';
+import * as Location from 'expo-location';
+import axios from 'axios';
+import EventsMap from './components/EventsMap';
+import { Event } from './interfaces/types'; // Import the Event type
 
 export default function App() {
+  const [location, setLocation] = useState<{
+    latitude: number;
+    longitude: number;
+    latitudeDelta: number;
+    longitudeDelta: number;
+  } | null>(null);
+  const [events, setEvents] = useState<Event[]>([]);
+
+  useEffect(() => {
+    const getPermissionAndLocation = async () => {
+      let { status } = await Location.requestForegroundPermissionsAsync();
+      if (status !== 'granted') {
+        console.log('Permission denied');
+        return;
+      }
+      const currentLocation = await Location.getCurrentPositionAsync({});
+      setLocation({
+        latitude: currentLocation.coords.latitude,
+        longitude: currentLocation.coords.longitude,
+        latitudeDelta: 0.01,
+        longitudeDelta: 0.01,
+      });
+      console.log('Got permission forLocation:', currentLocation);
+    };
+
+    getPermissionAndLocation();
+  }, []);
+
+  useEffect(() => {
+    const fetchEventsByLocation = async (latitude: number, longitude: number) => {
+      const root_URL = 'http://192.168.100.39:3000/api/';
+      const GET_EVENTS_API_URL = root_URL + 'events/';
+      try {
+        const response = await axios.get<Event[]>(GET_EVENTS_API_URL, {
```

Comment:
I would suggest implementing some kind of EventService/ RestCallService in the future that can handle the actual requests and responses (with axios) and avoid doing everything in this component.

Ex file: /api/EventService.js or /api/EventService.ts

import axios from "axios";
const API_URL = "http://localhost:8080/api/events/";

export async function saveEvent(event) {
    return await axios.post(API_URL, event);
}

export async function getEventById(id) {
    return await axios.get(`${API_URL}/${id}`);
}
...etc.